openrpc_validator/src/lib.rs

```diff
@@ -115,23 +115,39 @@ impl FireboltOpenRpc {
         None
     }
 
+    fn resolve_schema_properties(
+        &self,
+        result_schema_map: &Map<String, Value>,
+        rpc_method: &RpcMethod,
+    ) -> Map<String, Value> {
+        // Check if the result contains a $ref to an object type and if so, resolve it.
+        if let Some(result_ref_schema) = self.get_result_ref_schemas(result_schema_map) {
+            return result_ref_schema;
+        }
+
+        // The result wasn't an object reference, determine the type and return the result property.
+        let mut result_map = Map::new();
+        let result_value = result_schema_map.get("type").unwrap_or(&Value::Null);
+        result_map.insert(rpc_method.result.name.clone(), result_value.clone());
+        result_map
+    }
+
     pub fn get_result_properties_schema_by_name(&self, name: &str) -> Option<Map<String, Value>> {
         if let Some(method) = self.get_method_by_name(name) {
             if let Some(result_schema_map) = method.result.schema.as_object() {
                 if let Some(any_of_map) = result_schema_map.get("anyOf") {
+                    // FIXME: This will always return the result schema for the first object in the anyOf array, because we don't have a way to determine
```

Comment:
With the current firebolt schemas we never get into this situation where we need to access nested result schemas like this, but still needs more discussion on how we can support this in the future. Had prior discussions related to this with KP and the conclusion is that it's something that needs figuring out. I'll re-open that dialog.